runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/status/WorkerStatusPagesTest.java

```diff
@@ -38,12 +39,13 @@
   private final Server server = new Server();
   private final LocalConnector connector = new LocalConnector(server);
   @Mock private MemoryMonitor mockMemoryMonitor;
+  private final BooleanSupplier mockHealthyIndicator = () -> true;
   private WorkerStatusPages wsp;
 
   @Before
   public void setUp() throws Exception {
     MockitoAnnotations.initMocks(this);
-    wsp = new WorkerStatusPages(server, mockMemoryMonitor);
+    wsp = new WorkerStatusPages(server, mockMemoryMonitor, mockHealthyIndicator);
```

Comment:
Should we also test the negative case (i.e. indicator being false)

Reply:
Thanks for the suggestion. I will take a look.

Reply:
Maybe a future PR, but it might be good to have an end to end test for the recovery.

Reply:
Sure. I will add the unit test here later. Adding an integration test will requires more work so I will defer it to a later PR.

Reply:
Unit test done.